apps/api/src/workspace-role/service/workspace-role.service.ts

```diff
@@ -94,6 +98,45 @@ export class WorkspaceRoleService {
             }
           }
         },
+        select: {
+          id: true
+        }
+      })
+    )
+
+    if (dto.projectEnvironments) {
+      // Create the project associations
+      const projectSlugToIdMap = await this.getProjectSlugToIdMap(
+        dto.projectEnvironments
+          .map((pe) => pe.projectSlug)
+          .filter((slug) => slug)
+      )
+
+      for (const pe of dto.projectEnvironments) {
+        const projectId = projectSlugToIdMap.get(pe.projectSlug)
+        if (projectId) {
+          // Create the project workspace role association with the environments accessible on the project
+          op.push(
+            this.prisma.projectWorkspaceRoleAssociation.create({
+              data: {
+                roleId: workspaceRoleId,
+                projectId: projectId,
+                environments: {
+                  connect: pe.environmentSlugs.map((slug) => ({ slug }))
```

Comment:
For every environment slug belonging to a project, we need to make sure that the environment does indeed belong to a project.